x-pack/plugins/actions/server/builtin_action_types/lib/connector_token_client.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { omitBy, isUndefined } from 'lodash';
+import { ConnectorToken } from '../../types';
+import { EncryptedSavedObjectsClient } from '../../../../encrypted_saved_objects/server';
+import {
+  Logger,
+  SavedObjectsClientContract,
+  SavedObjectsUtils,
+} from '../../../../../../src/core/server';
+import { CONNECTOR_TOKEN_SAVED_OBJECT_TYPE } from '../../constants/saved_objects';
+
+export const MAX_TOKENS_RETURNED = 1;
+
+interface ConstructorOptions {
+  encryptedSavedObjectsClient: EncryptedSavedObjectsClient;
+  unsecuredSavedObjectsClient: SavedObjectsClientContract;
+  logger: Logger;
+}
+
+interface CreateOptions {
+  connectorId: string;
+  token: string;
+  expiresAt: string;
+  tokenType?: string;
+}
+
+export interface UpdateOptions {
+  id: string;
+  token: string;
+  expiresAt: string;
+  tokenType?: string;
+}
+
+export class ConnectorTokenClient {
+  private readonly logger: Logger;
+  private readonly unsecuredSavedObjectsClient: SavedObjectsClientContract;
+  private readonly encryptedSavedObjectsClient: EncryptedSavedObjectsClient;
+
+  constructor({
```

Comment:
Should we add a `delete` function and clean up these connector tokens SOs when the connector SO is deleted? Is there any reason we would want to keep them around if a connector is deleted?